audformat/core/attachment.py

```diff
@@ -59,6 +60,67 @@ def __init__(
         self.path = path
         r"""Attachment path"""
 
+    @property
+    def files(
+            self,
+    ) -> typing.List[str]:
+        r"""List all files part of the attachment.
+
+        List recursively the relative path
+        of all files that exist
+        under :attr:`audformat.Attachment.path`
+        on hard disc.
+
+        Raises:
+            FileNotFoundError: if a path
+                associated with an attachment
+                cannot be found
+            RuntimeError: if any file
+                associated with the attachment
+                is a symlink
+            RuntimeError: if attachment is not part
+                of a database
+            RuntimeError: if database is not saved to disk
+
+        """
+        if not self._db:
+            raise RuntimeError(
+                'The attachment needs to be assigned to a database '
+                'before attached files can be listed.'
+            )
+        if not self._db.root:
+            raise RuntimeError(
+                'The database needs to be saved to disk '
+                'before attachment files can be listed.'
+            )
+        self._check_path(self._db.root)
+
+        files = []
+
+        path = audeer.path(self._db.root, self.path)
+        if os.path.isdir(path):
+            files = audeer.list_file_names(
+                path,
+                recursive=True,
+                hidden=True,
+            )
+            for file in files:
+                if os.path.islink(file):
```

Comment:
Will this catch the case where the user creates a symlink to a folder?

Reply:
We cannot get this case as symlinks to folders are not returned by `audeer.list_file_names()`.

Reply:
But I will add another test case where a file is inside a symlinked folder.